src/Abs.hs

```diff
@@ -16,8 +16,22 @@ import UnliftIO
 -- This is the file where we define types to avoid cyclic dependencies.
 -- No project file with actual code should be imported from here.
 
-
-data Val = VStr String
+data Expr
+  = Var Text
+  | Lit Text
+  | Int Integer
+  | Negation Expr
+  | Sum      Expr Expr
+  | Subtr    Expr Expr
+  | Product  Expr Expr
+  deriving (Eq, Ord, Show)
+
+data Val = VStr Text | VInt Integer deriving (Show)
+
+instance Num Val where
```

Comment:
This instance declaration raises a warning.

Reply:
We probably don't want to have this instance anyway, as there is no way to handle type errors. This provided we want to do dynamic typing of course.